lib/liquid/standardfilters.rb

```diff
@@ -188,15 +188,11 @@ def reverse(input)
 
     # map/collect on a given property
     def map(input, property)
-      InputIterator.new(input).map do |e|
-        e = e.call if e.is_a?(Proc)
-
-        if property == "to_liquid".freeze
-          e
-        elsif e.respond_to?(:[])
-          r = e[property]
-          r.is_a?(Proc) ? r.call : r
-        end
+      ary = InputIterator.new(input).map { |e| e.is_a?(Proc) ? e.call : e }
+      if property == "to_liquid".freeze
+        ary
+      elsif ary.first.respond_to?(:[])
+        ary.map { |e| (e = get_property(e, property)).is_a?(Proc) ? e.call : e }
```

Comment:
Could we implement this without a double `map`? This change is making the method slower.

Reply:
@Thibaut I was trying to mirror the format of the other methods. Is the time complexity that much different because it still runs linearly right?

Reply:
The bigger problem here is that this allocates an extra array now. This should be done without the extra array.